engine/src/main/java/org/terasology/rendering/nui/SortOrder.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2018 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.rendering.nui;
+
+import com.google.common.collect.Queues;
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.subsystem.config.BindsManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.input.BindButtonEvent;
+import org.terasology.input.BindButtonSubscriber;
+import org.terasology.input.BindableButton;
+import org.terasology.input.Keyboard;
+import org.terasology.input.internal.BindableButtonImpl;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.registry.In;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.Map;
+
+
+@RegisterSystem(RegisterMode.ALWAYS)
+public class SortOrder extends BaseComponentSystem {
+    private static int current;
+    private static ArrayList<Integer[]> layersFilled; //arg1 of the Integer[] is the layer depth, arg2 is the number of things on that layer
+    private static int index;
+    private static boolean inSortOrder;
+    private static ArrayList<CoreScreenLayer> enabledWidgets;
+    private static boolean initialized = false;
+    private static ArrayList<Integer> used;
+
+    @In
+    private BindsManager bindsManager;
+
+    /**
+     * Initializes sort order
+     * @param event
+     * @param player
+     */
+    @ReceiveEvent
+    public void onPlayerSpawnedEvent(OnPlayerSpawnedEvent event, EntityRef player) {
+        initialized = true;
+        Map<Integer, BindableButton> keys = bindsManager.getKeyBinds();
+        if (keys.containsKey(Keyboard.Key.SEMICOLON.getId())) {
+            keys.get(Keyboard.Key.SEMICOLON.getId()).subscribe(new BindButtonSubscriber() {
```

Comment:
Semi colon might be good for testing, but not very intuitive - maybe we could use something more like alt-tabbing. Some other games and apps use ctrl-tab or shift-tab, think that would be possible? Maybe even both - shift being an order reverser.

Reply:
I'd be curious if we could suppress the online players view if any modifier key is also used, but that could be something to worry about later :-)